src/ell/studio/config.py

```diff
@@ -10,31 +11,36 @@
 
 # todo. maybe we default storage dir and other things in the future to a well-known location
 # like ~/.ell or something
-@lru_cache
+@lru_cache(maxsize=1)
 def ell_home() -> str:
     return os.path.join(os.path.expanduser("~"), ".ell")
 
 
 class Config(BaseModel):
     pg_connection_string: Optional[str] = None
     storage_dir: Optional[str] = None
-
-    @classmethod
-    def create(
-        cls,
-        storage_dir: Optional[str] = None,
-        pg_connection_string: Optional[str] = None,
-    ) -> 'Config':
-        pg_connection_string = pg_connection_string or os.getenv("ELL_PG_CONNECTION_STRING")
-        storage_dir = storage_dir or os.getenv("ELL_STORAGE_DIR")
-
+    mqtt_connection_string: Optional[str] = None
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def model_post_init(self, __context):
+        # Storage
+        self.pg_connection_string = self.pg_connection_string or os.getenv(
+            "ELL_PG_CONNECTION_STRING")
+        self.storage_dir = self.storage_dir or os.getenv("ELL_STORAGE_DIR")
+        
         # Enforce that we use either sqlite or postgres, but not both
-        if pg_connection_string is not None and storage_dir is not None:
+        if self.pg_connection_string is not None and self.storage_dir is not None:
             raise ValueError("Cannot use both sqlite and postgres")
-        
+
         # For now, fall back to sqlite if no PostgreSQL connection string is provided
-        if pg_connection_string is None and storage_dir is None:
+        if self.pg_connection_string is None and self.storage_dir is None:
             # This intends to honor the default we had set in the CLI
-            storage_dir = os.getcwd()
+            # todo. better default?
+            self.storage_dir = os.getcwd()
```

Comment:
I’m ending up with SQLite dbs in various places. Maybe we standardize on ell home as a default?